### ISSUE TITLE
Uncaught TypeError: Invalid value used in weak set after throw an error

### ISSUE THREAD

**Issue by Dexus** [GoogleChromeLabs/comlink#409]
I get this error:

console.trace:
```
worker.js?c326:19 Key to short or to long
chacha20poly1305_encrypt @ worker.js?c326:19
callback @ comlink.mjs?3d15:90
```
error:
```
Uncaught (in promise) TypeError: Invalid value used in weak set
    at WeakSet.add (<anonymous>)
    at eval (comlink.mjs?3d15:119)
```

while I use:

worker.js:
```
import * as Comlink from 'comlink';
import("../../rust-src/pkg").then(wasm => {
  console.log("Worker started");
  wasm.init();
  return {
    request() {
      return wasm.request()
    },

    chacha20poly1305_decrypt(key, noce, str) {
      return wasm.chacha20poly1305_decrypt(key, noce, str).catch(console.error)
    },

    chacha20poly1305_encrypt(key, noce, str) {
      let resp;
      try {
        resp = wasm.chacha20poly1305_encrypt(key, noce, str)
      } catch (err) {
        console.trace(err);
        throw err;
      }
      return resp
    },

    aes_encrypt(str) {
      return wasm.aes_encrypt(str)
    },

    hash(str) {
      return wasm.hash(str)
    },
  }


}).then((MyClassObj) => {
  Comlink.expose(MyClassObj);
});
```

this affects(PR #382,#384)
https://github.com/GoogleChromeLabs/comlink/blob/master/src/comlink.ts#L178

I currently use: npm ` "comlink": "^4.1.0",`

**Comment by Dexus** [comment]
okay solution of the problem throw the error as error again.

**Comment by felixfbecker** [comment]
@surma I noticed this while reading the code today - if code throws a scalar (string or number), comlink will throw because it tries to add the value to a WeakSet, which only supports objects. I think this should be reopened?

**Comment by surma** [comment]
@felixfbecker Can you come up with a reproduction case? Then I can add it to the tests and fix it.

**Comment by felixfbecker** [comment]
Sure, here ya go: https://github.com/GoogleChromeLabs/comlink/pull/453

Note: I am not actually personally affected by this, just noticed it from reading the code ✌️

**Comment by surma** [comment]
Fixed and published in v4.2.1
